dbms/src/Interpreters/ExpressionAnalyzer.h

```diff
@@ -219,6 +219,12 @@ class SelectQueryExpressionAnalyzer : public ExpressionAnalyzer
       */
     void tryMakeSetForIndexFromSubquery(const ASTPtr & subquery_or_table_name);
 
+    /**
+    * Checks if subquery is not a plain StorageSet.
+    * Because while making set we will read data from StorageSet which is not allowed.
+    */
+    bool allowSetForIndexFromSubquery(const ASTPtr & subquery_of_table_name);
```

Comment:
Should we name it something like `isNotSet`? though I'd use `isSet` and negate later.